### PR TITLE
22988 - Fixed the Amalgamation Corp Type Bug

### PR DESCRIPTION
**Pull request by ArwenQin**
*Issue #:* /bcgov/entity#22988

*Description of changes:*
- Added the computed corp type when primary business is continued in

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-728-rdomavzf.web.app

**Comment by ArwenQin** [comment]
Try:
https://business-create-dev--pr-728-rdomavzf.web.app/amalg-short-information?id=TQMmue0hhd&accountid=3040

File: package.json
Comment (severinbeauvais) [inline package.json:1]: While you're in here, can you please:
- npm uninstall @volar-plugins/vetur
- npm install --save-dev volar-service-vetur

Comment (ArwenQin) [reply package.json:1]: Will do!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-728-rdomavzf.web.app

**Comment by ArwenQin** [comment]
@severinbeauvais added unit tests and updated the packages. Is it ok to merge now?
Thanks!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-728-rdomavzf.web.app

File: tests/unit/amalgamation-mixin.spec.ts [hunk 1]
```diff
@@ -286,4 +286,24 @@ describe('Amalgamation Mixin - rules', () => {
...
+    const setEntityTypeSpy = vi.spyOn(wrapper.vm, 'setEntityType')
+    const testCases = [
+      { input: CorpTypeCd.CONTINUE_IN, expected: CorpTypeCd.BC_COMPANY },
+      { input: CorpTypeCd.BEN_CONTINUE_IN, expected: CorpTypeCd.BENEFIT_COMPANY },
+      { input: CorpTypeCd.CCC_CONTINUE_IN, expected: CorpTypeCd.BC_CCC },
+      { input: CorpTypeCd.ULC_CONTINUE_IN, expected: CorpTypeCd.BC_ULC_COMPANY },
+      { input: CorpTypeCd.BC_ULC_COMPANY, expected: CorpTypeCd.BC_ULC_COMPANY },
+      { input: CorpTypeCd.BC_COMPANY, expected: CorpTypeCd.BC_COMPANY },
+      { input: CorpTypeCd.BENEFIT_COMPANY, expected: CorpTypeCd.BENEFIT_COMPANY }
+    ]
```
Comment (severinbeauvais) [inline tests/unit/amalgamation-mixin.spec.ts:300]: You're missing one for BC_CCC ;)

Comment (severinbeauvais) [reply tests/unit/amalgamation-mixin.spec.ts:300]: You can merge after fixing this.

Comment (ArwenQin) [reply tests/unit/amalgamation-mixin.spec.ts:300]: Will do, thanks a lot

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-728-rdomavzf.web.app

**Pull request merged**